apps/E2E/src/ButtonV1/pages/ButtonV1PageObject.ts

```diff
@@ -10,17 +10,23 @@ class ButtonV1PageObject extends BasePage {
   /******************************************************************/
   /**************** UI Element Interaction Methods ******************/
   /******************************************************************/
-  async didOnClickCallbackFire(): Promise<boolean> {
-    const callbackText = await By(BUTTON_ON_PRESS);
-    await browser.waitUntil(async () => await callbackText.isDisplayed(), {
-      timeout: this.waitForUiEvent,
-      timeoutMsg: 'The OnClick callback did not fire.',
-      interval: 1000,
-    });
-
+  async waitForOnClickCallbackToFire(errorMsg: string): Promise<boolean> {
+    const callbackText = await this._callbackText;
+    await this.waitForCondition(async () => await callbackText.isDisplayed(), errorMsg);
     return await callbackText.isDisplayed();
   }
 
+  async resetTest(): Promise<void> {
```

Comment:
Shouldn't this also apply to ButtonLegacy?

Reply:
It should. Added the same hook in its test as well.